### PR TITLE
[PhpUnitBridge] Allow phpunit install --ignore-platform-reqs to test PHP 8.0

### PR DESCRIPTION
**Pull request by mazarini**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| New feature?  | yes 
| Tickets  |  #33592
| License       | MIT

Allow phpunit install --ignore-platform-reqs to test php 8.0

### CODE REVIEW

**Comment by mazarini** [comment]
All seems to work before a bad commit on src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php
and a try to resolve conficts.
I can nothing nothing for DeprecationErrorHandler.php (fix php syntax only to test)

I have to look on simple-phpunit.php to understand what append.

You can see https://travis-ci.org/mazarini/demo/builds/587072348 : symfony/demo working with php 8.0

**Comment by mazarini** [comment]
Now I need to solve the reinstall of phpunit each time. Its come from bad resolving confict.
Second problem is "SYMFONY_PHPUNIT_BRIDGE_PR=ticket_33592".

Note that now symfony/demo is working with php 8.0 on TravisCI.

**Comment by mazarini** [comment]
I'm not sure that I choose the best solution with using --ignore-platform-reqs. 
Using an EnvVar can do the same and allow what option we want, not only this one.

File: .travis.yml [hunk 1]
```diff
@@ -195,7 +195,7 @@ install:
       # To run a PR with a patched phpunit-bridge, first submit the path for the
       # phpunit-bridge as a separate PR against the next feature-branch then
       # uncomment and update the following line with that PR number
-      #SYMFONY_PHPUNIT_BRIDGE_PR=32886
+      SYMFONY_PHPUNIT_BRIDGE_PR=ticket_33592
```
Comment (stof) [inline .travis.yml]: This change is wrong:

- `ticket_33592` is not a valid PR number
- this PR is a PR editing the bridge itself, not a PR when you want to use a pending PR on the bridge to run tests, so there is no reason to use that feature.

**Review by stof — CHANGES_REQUESTED** [review]
Why are you reincluding diff of other features in this PR ? I think you messed the git history of your PR.

File: CHANGELOG-4.3.md [hunk 2]
```diff
@@ -569,4 +569,4 @@ To get the diff between two versions, go to https://github.com/symfony/symfony/c
  * feature #28976 [DI] Add a "default" EnvProcessor (jderusse)
  * feature #29127 [DomCrawler] Added return of element name in `extract()` method (andrey-helldar)
  * feature #29145 [Workflow] Trigger `entered` event for subject entering in the Workflow for the first time (lyrixx)
-
+ * feature #33592 [PhpUnitBridge] allow install of PHPUnit with option --ignore-platform-reqs (mazarini)
```
[the remark this answers is not in the document]
Comment (mazarini) [reply CHANGELOG-4.3.md]: I think it's better to do a new PR. I have make to much error (problem with php-cs, apply patch, try to solve conflict). 
Tell me the branch to fork.
And choose between "phpunit --ignore-platform-reqs" or "export PHPUNIT_COMPOSER_FLAG = --ignore-platform-reqs && phpunit"

File: src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php [hunk 3]
```diff
@@ -78,11 +79,11 @@
     $PHP .= ' -qrr';
 }
 
-$defaultEnvs = array(
+$defaultEnvs = [
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php]: please revert all changes from fabbot, they're not applicable to this particular bridge

**Comment by nicolas-grekas** [comment]
Note that there is an alternative since this is for early experimentations:
- run `simple-phpunit install` on PHP 7.*
- run it again with proper arguments on PHP 8

**Comment by mazarini** [comment]
I hope file are well reset now.
I have use an envvar ($SYMFONY_PHPUNIT_COMPOSER_OPTION). 
We can use "--ignore-platform-reqs" or an other option.

**Comment by alexander-schranz** [comment]
@mazarini are you still working on this? Looking forward to this feature to test agains PHP 8.

**Comment by mazarini** [comment]
@alexander-schranz I'm waiting feedback to know what I have to do : use flag or envvar, name of flag or anvvar. I prefere envvar.       
I'll probably close it and do a new proposal when 4.4 is ready.
Note I use it on travis, just copy the modified file before test.

**Comment by mazarini** [comment]
Replace by PR #34615

**Pull request closed without merging**